clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -951,7 +978,13 @@ private Record nextFetchedRecord() {
                         return null;
                     }
                     currentBatch = batches.next();
+
                     maybeEnsureValid(currentBatch);
+
+                    if (isolationLevel == IsolationLevel.READ_COMMITTED && isBatchAborted(currentBatch)) {
+                        continue;
```

Comment:
I added the code to update the offset of the skipped batches and added a test for this as well.